// @growthbunker/vueflags/dist/vuesocial.ssr.js
'use strict';Object.defineProperty(exports,'__esModule',{value:true});var vuedarkmode=require('@growthbunker/vuedarkmode');/**************************************************************************
 * MIXINS > THEME
 * @docs https://vuejs.org/v2/guide/mixins.html
 ***************************************************************************/

var ThemeMixin = {
  props: {
    theme: {
      type: String,
      default: null,
      validator: function validator(x) {
        return ["dark", "light"].includes(x)
      }
    }
  },

  computed: {
    computedTheme: function computedTheme() {
      if (this.theme) {
        return this.theme
      }

      return this.$gb.vuesocial.theme
    }
  }
};//

var script = {
  components: {
    BaseSpinner: vuedarkmode.BaseSpinner
  },

  mixins: [ThemeMixin],

  props: {
    iconPath: {
      type: String,
      default: null
    },
    iconTheme: {
      type: String,
      default: "color",
      validator: function validator(x) {
        return ["black", "color", "white"].includes(x)
      }
    },
    disabled: {
      type: Boolean,
      default: false
    },
    fullWidth: {
      type: Boolean,
      default: false
    },
    label: {
      type: String,
      default: null
    },
    loading: {
      type: Boolean,
      default: false
    },
    network: {
      type: String,
      required: true,
      validator: function validator(x) {
        return [
          "500px",
          "airbnb",
          "amazon",
          "android",
          "apple",
          "bankin",
          "behance",
          "bitly",
          "blackberry",
          "blogger",
          "buffer",
          "chrome",
          "codepen",
          "dailymotion",
          "dribbble",
          "drive",
          "dropbox",
          "envato",
          "evernote",
          "facebook",
          "fancy",
          "feedly",
          "firefox",
          "flickr",
          "foursquare",
          "github",
          "google",
          "hangout",
          "instagram",
          "internetexplorer",
          "invision",
          "linkedin",
          "magento",
          "medium",
          "messenger",
          "opera",
          "paypal",
          "periscope",
          "photoshop",
          "pinterest",
          "pocket",
          "principle",
          "producthunt",
          "rdio",
          "reddit",
          "rss",
          "safari",
          "scoopit",
          "shopify",
          "sketch",
          "skype",
          "slack",
          "snapchat",
          "soundcloud",
          "spotify",
          "stackoverflow",
          "tinder",
          "trello",
          "tumblr",
          "twitter",
          "viadeo",
          "viber",
          "vimeo",
          "vine",
          "whatsapp",
          "windowsphone",
          "wordpress",
          "yelp",
          "youtube"
        ].includes(x)
      }
    },
    reverse: {
      type: Boolean,
      default: false
    },
    rounded: {
      type: Boolean,
      default: false
    },
    size: {
      type: String,
      default: "default",
      validator: function validator(x) {
        return ["nano", "micro", "mini", "small", "default", "medium", "large"].includes(x)
      }
    }
  },

  computed: {
    logo: function logo() {
      var path = this.$gb.vuesocial.iconPath || this.iconPath || "";

      // Remove last character if slash
      path = path.replace(/\/$/, "");

      return (path + "/" + (this.network) + "_" + (this.iconTheme) + ".svg")
    },

    spinnerSize: function spinnerSize() {
      if (["nano", "micro"].includes(this.size)) {
        return "nano"
      } else if (["mini", "small", "default"].includes(this.size)) {
        return "mini"
      }

      return "default"
    }
  },

  methods: {
    // --> EVENT LISTENERS <--

    onClick: function onClick(event) {
      this.$emit("click", this.network, event);
    }
  }
};function normalizeComponent(template, style, script, scopeId, isFunctionalTemplate, moduleIdentifier /* server only */, shadowMode, createInjector, createInjectorSSR, createInjectorShadow) {
    if (typeof shadowMode !== 'boolean') {
        createInjectorSSR = createInjector;
        createInjector = shadowMode;
        shadowMode = false;
    }
    // Vue.extend constructor export interop.
    var options = typeof script === 'function' ? script.options : script;
    // render functions
    if (template && template.render) {
        options.render = template.render;
        options.staticRenderFns = template.staticRenderFns;
        options._compiled = true;
        // functional template
        if (isFunctionalTemplate) {
            options.functional = true;
        }
    }
    // scopedId
    if (scopeId) {
        options._scopeId = scopeId;
    }
    var hook;
    if (moduleIdentifier) {
        // server build
        hook = function (context) {
            // 2.3 injection
            context =
                context || // cached call
                    (this.$vnode && this.$vnode.ssrContext) || // stateful
                    (this.parent && this.parent.$vnode && this.parent.$vnode.ssrContext); // functional
            // 2.2 with runInNewContext: true
            if (!context && typeof __VUE_SSR_CONTEXT__ !== 'undefined') {
                context = __VUE_SSR_CONTEXT__;
            }
            // inject component styles
            if (style) {
                style.call(this, createInjectorSSR(context));
            }
            // register component module identifier for async chunk inference
            if (context && context._registeredComponents) {
                context._registeredComponents.add(moduleIdentifier);
            }
        };
        // used by ssr in case component is cached and beforeCreate
        // never gets called
        options._ssrRegister = hook;
    }
    else if (style) {
        hook = shadowMode
            ? function (context) {
                style.call(this, createInjectorShadow(context, this.$root.$options.shadowRoot));
            }
            : function (context) {
                style.call(this, createInjector(context));
            };
    }
    if (hook) {
        if (options.functional) {
            // register for functional component in vue file
            var originalRender = options.render;
            options.render = function renderWithStyleInjection(h, context) {
                hook.call(context);
                return originalRender(h, context);
            };
        }
        else {
            // inject component registration as beforeCreate hook
            var existing = options.beforeCreate;
            options.beforeCreate = existing ? [].concat(existing, hook) : [hook];
        }
    }
    return script;
}function createInjectorSSR(context) {
    if (!context && typeof __VUE_SSR_CONTEXT__ !== 'undefined') {
        context = __VUE_SSR_CONTEXT__;
    }
    if (!context)
        { return function () { }; }
    if (!('styles' in context)) {
        context._styles = context._styles || {};
        Object.defineProperty(context, 'styles', {
            enumerable: true,
            get: function () { return context._renderStyles(context._styles); }
        });
        context._renderStyles = context._renderStyles || renderStyles;
    }
    return function (id, style) { return addStyle(id, style, context); };
}
function addStyle(id, css, context) {
    var group =  css.media || 'default' ;
    var style = context._styles[group] || (context._styles[group] = { ids: [], css: '' });
    if (!style.ids.includes(id)) {
        style.media = css.media;
        style.ids.push(id);
        var code = css.source;
        style.css += code + '\n';
    }
}
function renderStyles(styles) {
    var css = '';
    for (var key in styles) {
        var style = styles[key];
        css +=
            '<style data-vue-ssr-id="' +
                Array.from(style.ids).join(' ') +
                '"' +
                (style.media ? ' media="' + style.media + '"' : '') +
                '>' +
                style.css +
                '</style>';
    }
    return css;
}/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('button',{class:[
    "gb-social-button",
    "gb-social-button--" + _vm.size,
    "gb-social-button--" + _vm.computedTheme,
    {
      "gb-social-button--disabled": _vm.disabled || _vm.loading,
      "gb-social-button--full-width": _vm.fullWidth,
      "gb-social-button--loading": _vm.loading,
      "gb-social-button--not-reverse": !_vm.reverse,
      "gb-social-button--reverse": _vm.reverse,
      "gb-social-button--rounded": _vm.rounded
    }
  ],attrs:{"disabled":_vm.disabled,"type":"button"},on:{"click":_vm.onClick}},[_vm._ssrNode("<span class=\"gb-social-button__focuser\"></span><span class=\"gb-social-button__inner\"><img"+(_vm._ssrAttr("src",_vm.logo))+" class=\"gb-social-button__logo\">"+((_vm.label)?("<span class=\"gb-social-button__label\">"+(_vm._s(_vm.label))+"</span>"):"<!---->")+"</span>"),(_vm.loading)?_c('base-spinner',{staticClass:"gb-social-button__spinner",attrs:{"size":_vm.spinnerSize,"color":"black"}}):_vm._e()],2)};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = function (inject) {
    if (!inject) { return }
    inject("data-v-19ceaec0_0", { source: ".gb-social-button{position:relative;display:inline-block;outline:0;border-width:1px;border-style:solid;background-position:center;font-family:Heebo,\"Helvetica Neue\",Source Sans Pro,Helvetica,Arial,sans-serif;transition:all 250ms linear;user-select:none;cursor:pointer}.gb-social-button .gb-social-button__focuser{position:absolute;top:-4px;right:-4px;bottom:-4px;left:-4px;border-width:1px;border-style:solid;opacity:0;transition:all linear 250ms}.gb-social-button .gb-social-button__inner{display:flex;align-items:center;justify-content:center}.gb-social-button .gb-social-button__inner .gb-social-button__logo{display:inline-block}.gb-social-button .gb-social-button__inner .gb-social-button__label{margin-left:6px;font-weight:500}.gb-social-button--nano{padding:3px 6px;border-radius:2px;font-size:10px;line-height:8px}.gb-social-button--nano .gb-social-button__focuser{border-radius:4px}.gb-social-button--nano .gb-social-button__inner .gb-social-button__logo{width:10px;height:10px}.gb-social-button--nano .gb-social-button__inner .gb-social-button__label{margin-left:2px}.gb-social-button--micro{padding:5px 8px;border-radius:3px;font-size:11px;line-height:10px}.gb-social-button--micro .gb-social-button__focuser{border-radius:5px}.gb-social-button--micro .gb-social-button__inner .gb-social-button__logo{width:12px;height:12px}.gb-social-button--micro .gb-social-button__inner .gb-social-button__label{margin-left:3px}.gb-social-button--mini{padding:7px 10px;border-radius:4px;font-size:12px;line-height:12px}.gb-social-button--mini .gb-social-button__focuser{border-radius:6px}.gb-social-button--mini .gb-social-button__inner .gb-social-button__logo{width:14px;height:14px}.gb-social-button--mini .gb-social-button__inner .gb-social-button__label{margin-left:4px}.gb-social-button--small{padding:9px 12px;border-radius:5px;font-size:13px;line-height:14px}.gb-social-button--small .gb-social-button__focuser{border-radius:7px}.gb-social-button--small .gb-social-button__inner .gb-social-button__logo{width:16px;height:16px}.gb-social-button--small .gb-social-button__inner .gb-social-button__label{margin-left:5px}.gb-social-button--default{padding:11px 14px;border-radius:6px;font-size:14px;line-height:16px}.gb-social-button--default .gb-social-button__focuser{border-radius:8px}.gb-social-button--default .gb-social-button__inner .gb-social-button__logo{width:18px;height:18px}.gb-social-button--default .gb-social-button__inner .gb-social-button__label{margin-left:6px}.gb-social-button--medium{padding:13px 16px;border-radius:7px;font-size:15px;line-height:18px}.gb-social-button--medium .gb-social-button__focuser{border-radius:9px}.gb-social-button--medium .gb-social-button__inner .gb-social-button__logo{width:20px;height:20px}.gb-social-button--medium .gb-social-button__inner .gb-social-button__label{margin-left:7px}.gb-social-button--large{padding:15px 18px;border-radius:8px;font-size:16px;line-height:20px}.gb-social-button--large .gb-social-button__focuser{border-radius:10px}.gb-social-button--large .gb-social-button__inner .gb-social-button__logo{width:22px;height:22px}.gb-social-button--large .gb-social-button__inner .gb-social-button__label{margin-left:8px}.gb-social-button--dark{box-shadow:0 1px 5px 0 #18191a}.gb-social-button--dark.gb-social-button--not-reverse{border-color:#313d4f;background-color:#fff;color:#313d4f}.gb-social-button--dark.gb-social-button--not-reverse .gb-social-button__focuser{border-color:#fff}.gb-social-button--dark.gb-social-button--reverse{border-color:#fff;background-color:transparent;color:#fff}.gb-social-button--dark.gb-social-button--reverse:hover{background-color:#222c3c}.gb-social-button--dark.gb-social-button--reverse .gb-social-button__focuser{border-color:#fff}.gb-social-button--dark:hover{transform:translateY(-1px)}.gb-social-button--dark:active{transform:translateY(0)}.gb-social-button--dark:focus{transform:translateY(0)}.gb-social-button--light{box-shadow:0 1px 5px 0 #d3e4f0}.gb-social-button--light.gb-social-button--not-reverse{border-color:#c5d9e8;background-color:#fff;color:#2c405a}.gb-social-button--light.gb-social-button--not-reverse .gb-social-button__focuser{border-color:#c5d9e8}.gb-social-button--light.gb-social-button--reverse{border-color:#3f536e;background-color:transparent;color:#2c405a}.gb-social-button--light.gb-social-button--reverse:hover{background-color:#fff}.gb-social-button--light.gb-social-button--reverse .gb-social-button__focuser{border-color:#3f536e}.gb-social-button--light:hover{transform:translateY(-1px)}.gb-social-button--light:active{transform:translateY(0)}.gb-social-button--light:focus{transform:translateY(0)}.gb-social-button--disabled{opacity:.7;cursor:not-allowed}.gb-social-button--disabled .gb-social-button__inner{pointer-events:none}.gb-social-button--full-width{width:100%}.gb-social-button--loading{position:relative;opacity:1;cursor:wait}.gb-social-button--loading .gb-social-button__inner{opacity:0}.gb-social-button--loading .gb-social-button__spinner{position:absolute;top:50%;left:50%;transform:translate(-50%,-50%)}.gb-social-button--rounded{border-radius:100px}.gb-social-button--rounded .gb-social-button__focuser{border-radius:100px}.gb-social-button:active{background-size:100%;transition:background 0s}.gb-social-button:focus .gb-social-button__focuser{opacity:1}", map: undefined, media: undefined });

  };
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = "data-v-19ceaec0";
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject shadow dom */
  

  
  var __vue_component__ = normalizeComponent(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    false,
    undefined,
    createInjectorSSR,
    undefined
  );/**************************************************************************
 * IMPORTS
 ***************************************************************************/

/**************************************************************************
 * ENVIRONMENT CONFIGURATIONS
 ***************************************************************************/

// install function executed by Vue.use()
function install(Vue, options) {
  if (install.installed) {
    return
  } else {
    install.installed = true;
  }

  // Declare the component
  Vue.component("gb-social-button", __vue_component__);

  // Configure the theme to use (dark will always be the default theme)
  if (!Vue.prototype.$gb) {
    Vue.prototype.$gb = {};
  }

  Vue.prototype.$gb.vuesocial = {};
  Vue.prototype.$gb.vuesocial.iconPath = (options || {}).iconPath || null;
  Vue.prototype.$gb.vuesocial.theme = (options || {}).theme || "dark";
}

// Create module definition for Vue.use()
var plugin = {
  install: install
};

// To auto-install when vue is found
/* global window global */
var GlobalVue = null;

if (typeof window !== "undefined") {
  GlobalVue = window.Vue;
} else if (typeof global !== "undefined") {
  GlobalVue = global.Vue;
}

if (GlobalVue) {
  GlobalVue.use(plugin);
}exports.default=plugin;
